system/Helpers/array_helper.php

```diff
@@ -120,3 +120,40 @@ function _array_search_dot(array $indexes, array $array)
 		return $array[$currentIndex];
 	}
 }
+
+if (! function_exists('array_deep_search'))
+{
+	/**
+	 * Returns the value of an element with a key value in an array of uncertain depth.
+	 *
+	 * @param array  $array
+	 * @param string $key
+	 *
+	 * @return mixed|false
+	 */
+	function array_deep_search(array $array, string $key)
+	{
+		if (is_array($array))
```

Comment:
No need for the `is_array` check since it is already typehinted in the parameter.